tests/ast/data/interval.test

```diff
@@ -0,0 +1,81 @@
+## TEST CASE
+
+from snowflake.snowpark._internal.analyzer.expression import Interval
+
+df1 = session.create_dataframe(
+        [
+            [datetime.datetime(2010, 1, 1), datetime.datetime(2011, 1, 1)],
+            [datetime.datetime(2012, 1, 1), datetime.datetime(2013, 1, 1)],
+        ],
+        schema=["a", "b"],
+    )
+
+df2 = df1.select(
+    df1["a"]
+    + Column(
+        Interval(
```

Comment:
let's use make_interval here to test. Not sure we should support the `Column(Interval(...))` syntax at all.